eth/protocols/snap/sync.go

```diff
@@ -1960,6 +1987,17 @@ func (s *Syncer) forwardAccountTask(task *accountTask) {
 	}
 	it.Release()
 
+	// Persist the received account segements. These flat state maybe
+	// outdated during the sync, but it can be fixed later during the
+	// snapshot generation.
+	for i, hash := range res.hashes {
+		if task.needCode[i] || task.needState[i] {
+			break
+		}
```

Comment:
I don't follow the logic here. How come you don't `continue` instead of `break` ?

Reply:
I don't even understand why we even need to check `needCode` and `needState`. Could you explain?

Reply:
I think we can just YOLO-wite this to the snapshot -- what can go wong? At worst we'll have to overwrite it later

Reply:
Because in the `forwardAccountTask` there are two cases: (1) the account range is complete (2) the snap sync is interrupted because of pivot point moving. In the latter one we still want to persist the complete accounts, so the `needCode` and `needState` is the indicator of completeness here.

> I think we can just YOLO-wite this to the snapshot 

In theory yes, we can write whatever we have into the disk. But originally I thought it works more correctly to follow the existent logic

Reply:
For `needCode`, it simply means that the account has some code which we do not yet have locally, no? So we don't need to abort for that. 
Re `needState` doesn't that just mean that we don't have the storage trie yet? Regardless, we can just write it do disk anyway, we'll hopfully get the state later on.. ?

Reply:
Yes, I agree. For the flat states, we just write whatever we have. I'm fine to flush all states here.

Reply:
Fixed